packages/flutter/lib/src/services/text_input.dart

```diff
@@ -862,15 +862,12 @@ abstract class TextInputClient {
 ///    between the system's text input and a [TextInputClient].
 ///  * [EditableText], a [TextInputClient] that connects to and interacts with
 ///    the system's text input using a [TextInputConnection].
-class TextInputConnection {
-  TextInputConnection._(this._client)
+abstract class TextInputConnection {
```

Comment:
I was thinking that since the constructor was private before, nothing outside text_input.dart could have made TextInputConnection instances. Are there any other implications? Would it be better or safer to have non-functional stub implementations in TextInputConnection?

    void show() {
        assert(attached);
    }

There are a few public API changes, but my understanding is that none of them are breaking:

- TextInputConnection constructor made public
  - Needed for subclassing
- TextInputConnection gained three new methods: hide(), setClient(), clearClient()
  - The respective methods already exist in the "TextInput" method channel API
  - The method channel calls were previously made directly from TextInput